# 01 — Data Understanding

## Objective

Understand the structure, scope, and quality of the Decathlon retail dataset before performing any data cleaning or analysis.

## Dataset

- Source: [Kaggle](https://www.kaggle.com/datasets/anushka29/decathalon-data)
- File: Decathlon_Synthetic_Data_30000.xlsx
- Expected records: 30,000
- Expected columns: 39
- Expected period: 2024–2026m
## Analysis Scope

This notebook focuses only on understanding the raw dataset.

No data will be modified or cleaned in this stage.

In [2]:
import pandas as pd
import numpy as np

In [3]:
file_path = r"../data/raw/Decathlon_Synthetic_Data_30000.xlsx"
excel_file = pd.ExcelFile(file_path)
excel_file.sheet_names

['Sheet1']

In [4]:
df = pd.read_excel(file_path, sheet_name="Sheet1")

In [6]:
df.shape

(30000, 39)

In [10]:
df.columns.tolist()

['Order_ID',
 'Order_Date',
 'Order_Time',
 'Customer_ID',
 'Customer_Name',
 'Gender',
 'Age',
 'Age_Group',
 'City',
 'State',
 'Membership_Type',
 'Customer_Segment',
 'Product_ID',
 'Product_Name',
 'Product_Category',
 'Brand',
 'Sport_Type',
 'Quantity',
 'Unit_Price',
 'Discount_Percent',
 'Discount_Amount',
 'Sales_Amount',
 'Final_Amount',
 'Cost_Price',
 'Profit',
 'Store_ID',
 'Store_Name',
 'Sales_Channel',
 'Payment_Method',
 'Salesperson',
 'Delivery_Type',
 'Delivery_Days',
 'Customer_Rating',
 'Return_Status',
 'Return_Reason',
 'Promotion_Campaign',
 'Quarter',
 'Month',
 'Year']

In [7]:
df.head()

,Order_ID,Order_Date,Order_Time,Customer_ID,Customer_Name,Gender,Age,Age_Group,City,State,...,Salesperson,Delivery_Type,Delivery_Days,Customer_Rating,Return_Status,Return_Reason,Promotion_Campaign,Quarter,Month,Year
0,ORD100001,2024-12-07,11:53:06,CUST9177,Rohit Mehta,Female,61,Senior,Ahmedabad,Gujarat,...,Rahul Yadav,Home Delivery,6,3,Not Returned,Not Applicable,Diwali Mega Sale,Q4,December,2024
1,ORD100002,2024-02-20,20:12:38,CUST8120,Ananya Verma,Male,32,Adult,Chandigarh,Chandigarh,...,Arjun Mehta,Home Delivery,5,2,Not Returned,Not Applicable,New Year Fitness Campaign,Q1,February,2024
2,ORD100003,2024-09-15,14:17:10,CUST5601,Arjun Singh,Male,37,Adult,Kolkata,West Bengal,...,Rahul Mehta,Home Delivery,7,5,Not Returned,Not Applicable,Back To Sports,Q3,September,2024
3,ORD100004,2026-05-09,13:02:00,CUST4035,Karan Patel,Male,20,Adult,Jaipur,Rajasthan,...,Priya Verma,Pickup,0,5,Not Returned,Not Applicable,Back To Sports,Q2,May,2026
4,ORD100005,2024-01-22,21:47:28,CUST7304,Ananya Yadav,Female,17,Teen,Bengaluru,Karnataka,...,Pooja Singh,Home Delivery,4,5,Not Returned,Not Applicable,Independence Day Offer,Q1,January,2024


In [8]:
df.tail()

,Order_ID,Order_Date,Order_Time,Customer_ID,Customer_Name,Gender,Age,Age_Group,City,State,...,Salesperson,Delivery_Type,Delivery_Days,Customer_Rating,Return_Status,Return_Reason,Promotion_Campaign,Quarter,Month,Year
29995,ORD129996,2026-01-05,12:20:47,CUST6280,Aman Jain,Female,19,Teen,Jaipur,Rajasthan,...,Rohit Gupta,Pickup,0,1,Not Returned,Not Applicable,Summer Sale,Q1,January,2026
29996,ORD129997,2026-08-31,13:20:20,CUST9479,Riya Yadav,Male,30,Adult,Hyderabad,Telangana,...,Neha Sharma,Pickup,0,3,Not Returned,Not Applicable,Republic Day Sale,Q3,August,2026
29997,ORD129998,2024-05-28,16:47:29,CUST1668,Ananya Sharma,Female,14,Teen,Noida,Uttar Pradesh,...,Neha Sharma,Home Delivery,6,1,Not Returned,Not Applicable,New Year Fitness Campaign,Q2,May,2024
29998,ORD129999,2024-10-12,16:34:06,CUST8399,Arjun Malhotra,Male,13,Teen,Gurgaon,Haryana,...,Pooja Jain,Pickup,0,5,Not Returned,Not Applicable,Independence Day Offer,Q4,October,2024
29999,ORD130000,2024-07-05,11:45:13,CUST7696,Neha Patel,Female,44,Adult,Kolkata,West Bengal,...,Ananya Verma,Home Delivery,2,1,Not Returned,Not Applicable,Diwali Mega Sale,Q3,July,2024


In [11]:
df.dtypes

Order_ID                         str
Order_Date            datetime64[us]
Order_Time                       str
Customer_ID                      str
Customer_Name                    str
Gender                           str
Age                            int64
Age_Group                        str
City                             str
State                            str
Membership_Type                  str
Customer_Segment                 str
Product_ID                       str
Product_Name                     str
Product_Category                 str
Brand                            str
Sport_Type                       str
Quantity                       int64
Unit_Price                     int64
Discount_Percent               int64
Discount_Amount              float64
Sales_Amount                   int64
Final_Amount                 float64
Cost_Price                   float64
Profit                       float64
Store_ID                         str
Store_Name                       str
S

In [12]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 39 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Order_ID            30000 non-null  str           
 1   Order_Date          30000 non-null  datetime64[us]
 2   Order_Time          30000 non-null  str           
 3   Customer_ID         30000 non-null  str           
 4   Customer_Name       30000 non-null  str           
 5   Gender              30000 non-null  str           
 6   Age                 30000 non-null  int64         
 7   Age_Group           30000 non-null  str           
 8   City                30000 non-null  str           
 9   State               30000 non-null  str           
 10  Membership_Type     30000 non-null  str           
 11  Customer_Segment    30000 non-null  str           
 12  Product_ID          30000 non-null  str           
 13  Product_Name        30000 non-null  str           
 14  P

In [14]:
df.describe().T

,count,mean,min,25%,50%,75%,max,std
Order_Date,30000,2025-06-01 05:22:50.880000,2024-01-01 00:00:00,2024-09-15 00:00:00,2025-05-30 00:00:00,2026-02-08 00:00:00,2026-12-31 00:00:00,NaN
Age,30000.0,41.0507,13.0,27.0,41.0,56.0,70.0,16.612455
Quantity,30000.0,3.027533,1.0,2.0,3.0,4.0,8.0,1.367664
Unit_Price,30000.0,4607.466267,100.0,899.0,2999.0,6569.0,25111.0,4659.1416
Discount_Percent,30000.0,12.7244,0.0,6.0,13.0,19.0,40.0,8.127118
Discount_Amount,30000.0,1783.340965,0.0,89.9,749.7,2284.76,31801.0,2786.833574
Sales_Amount,30000.0,14019.160667,100.0,2499.0,8268.5,18999.0,138712.0,16881.930904
Final_Amount,30000.0,12235.819702,62.0,2124.15,7245.0,16620.25,129002.16,14853.93075
Cost_Price,30000.0,8478.924532,36.39,1439.0125,4957.33,11450.66,106431.41,10365.37868
Profit,30000.0,3756.89517,11.04,630.8975,2129.045,4911.89,43969.99,4874.749195


In [16]:
df.describe(include="object")

C:\Users\Himanshu\AppData\Local\Temp\ipykernel_23528\702825166.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include="object")


,Order_ID,Order_Time,Customer_ID,Customer_Name,Gender,Age_Group,City,State,Membership_Type,Customer_Segment,...,Store_Name,Sales_Channel,Payment_Method,Salesperson,Delivery_Type,Return_Status,Return_Reason,Promotion_Campaign,Quarter,Month
count,30000,30000,30000,30000,30000,30000,30000,30000,30000,30000,...,30000,30000,30000,30000,30000,30000,30000,30000,30000,30000
unique,30000,9121,8998,120,2,3,12,11,3,3,...,5,3,4,120,2,2,6,7,4,12
top,ORD100001,15:42:45,CUST4491,Rohit Jain,Female,Adult,Noida,Maharashtra,Basic,Returning,...,Decathlon Mumbai,Website,Wallet,Karan Sharma,Pickup,Not Returned,Not Applicable,Winter Sports Fest,Q1,May
freq,1,17,13,300,15164,20906,2650,5040,18153,13468,...,6112,10090,7800,347,20160,27584,27584,4578,7931,2725


In [17]:
df.isna().sum()

Order_ID              0
Order_Date            0
Order_Time            0
Customer_ID           0
Customer_Name         0
Gender                0
Age                   0
Age_Group             0
City                  0
State                 0
Membership_Type       0
Customer_Segment      0
Product_ID            0
Product_Name          0
Product_Category      0
Brand                 0
Sport_Type            0
Quantity              0
Unit_Price            0
Discount_Percent      0
Discount_Amount       0
Sales_Amount          0
Final_Amount          0
Cost_Price            0
Profit                0
Store_ID              0
Store_Name            0
Sales_Channel         0
Payment_Method        0
Salesperson           0
Delivery_Type         0
Delivery_Days         0
Customer_Rating       0
Return_Status         0
Return_Reason         0
Promotion_Campaign    0
Quarter               0
Month                 0
Year                  0
dtype: int64

In [20]:
df.duplicated().sum()

np.int64(0)

In [22]:
df['Order_Date'].min()

Timestamp('2024-01-01 00:00:00')

In [24]:
df['Order_Date'].max()

Timestamp('2026-12-31 00:00:00')

In [26]:
df['Year'].value_counts().sort_index()

Year
2024    10624
2025    10790
2026     8586
Name: count, dtype: int64

In [40]:
unique_orders_count = df['Order_ID'].nunique()
unique_customers_count = df['Customer_ID'].nunique()
unique_stores_count = df['Store_ID'].nunique()

pd.DataFrame([{'Orders':unique_orders_count,'Customer':unique_customers_count,'Stores':unique_stores_count}])


,Orders,Customer,Stores
0,30000,8998,5


In [41]:
summary = {
    "Rows": df.shape[0],
    "Columns": df.shape[1],
    "Unique Orders": df["Order_ID"].nunique(),
    "Unique Customers": df["Customer_ID"].nunique(),
    "Unique Products": df["Product_ID"].nunique(),
    "Unique Stores": df["Store_ID"].nunique(),
    "Duplicate Rows": df.duplicated().sum(),
    "Missing Cells": df.isna().sum().sum(),
    "Start Date": df["Order_Date"].min(),
    "End Date": df["Order_Date"].max()
}

summary

{'Rows': 30000,
 'Columns': 39,
 'Unique Orders': 30000,
 'Unique Customers': 8998,
 'Unique Products': 10,
 'Unique Stores': 5,
 'Duplicate Rows': np.int64(0),
 'Missing Cells': np.int64(0),
 'Start Date': Timestamp('2024-01-01 00:00:00'),
 'End Date': Timestamp('2026-12-31 00:00:00')}

## Initial Findings

The dataset contains 30,000 transaction records across 39 columns and covers the period from January 1, 2024 to December 31, 2026.

Each Order_ID is unique, resulting in 30,000 unique orders. The dataset contains 8,998 unique customers, 10 unique products, and 5 unique stores.

No duplicate rows or missing values were identified during the initial inspection.

The Order_Date column is already stored as a datetime data type, while numeric and categorical fields are appropriately recognized by pandas.

These findings describe the initial structure of the raw dataset. Business-rule validation and data-quality checks will be performed in the next stage.